### ISSUE TITLE
Unable to match cs:~foo/name revisions to cs:name revisions

### ISSUE THREAD

**Issue by stub42** [juju/charmstore#631]
I publish the PostgreSQL charm to cs:~postgresql-charmers/postgresql, and get back a revision like cs:~postgresql-charmers/postgresql-5. This is totally unrelated to the revision that end users will use, cs:postgresql-101. The end result being that when bug reports come in I have no idea what version of code was being run, unless I somehow maintain this mapping myself.

**Comment by frankban** [comment]
Yes the charm store needs to expose the link between entities and their promulgated revisions.

**Comment by johnsca** [comment]
As a temporary manual work around, you can use the `hash` field returned by `charm show` to find the right revision, if you already know which namespace is the promulgated one (which you can sort of figure out by the `perm` field):

``` bash
$ [[ `charm show cs:postgresql-101 hash` == `charm show cs:~postgresql-charmers/postgresql-5 hash` ]] && echo Match
Match
```
